stream_alert/rule_processor/rules_engine.py

```diff
@@ -21,18 +21,19 @@
 from stream_alert.rule_processor.threat_intel import StreamThreatIntel
 from stream_alert.shared import NORMALIZATION_KEY, resources
 from stream_alert.shared.alert import Alert
-from stream_alert.shared.rule import Rule
+from stream_alert.shared.rule import import_folders, Rule
 
 
 _IGNORE_KEYS = {StreamThreatIntel.IOC_KEY, NORMALIZATION_KEY}
 
 
 class RulesEngine(object):
     """Class to act as a rules engine that processes rules"""
-    def __init__(self, config):
+    def __init__(self, config, *rule_paths):
         """Initialize a RulesEngine instance to cache a StreamThreatIntel instance."""
         self._threat_intel = StreamThreatIntel.load_from_config(config)
         self._required_outputs_set = resources.get_required_outputs()
+        import_folders(*rule_paths)
```

Comment:
Since the `config` is passed in here and rule paths are computed directly from the config, do you think it should be the job of the `RulesEngine` to figure out the rule paths rather than the caller's responsibility?

Reply:
I had it this way before but changed it to this. I did so because I don't love that we pass a config object around in general and would rather stick to using primitive types like a basic string/list of strings.

I say this because there are places where we're passing around a whole config dict to end up only extracting/using one or two things from it.. and now for anything else to use that class properly, they'd have to construct the whole/valid config object to be passed to it. Thoughts on this? Happy to change back to the previous version.

Reply:
Hmm I see your point. In general, I'm happy with either, but maybe if we ever do a config rewrite we can work on better consistency

Reply:
++ for config rewrite +++++++